drake/core/Core.h

```diff
@@ -4,46 +4,112 @@
+    template <typename ScalarType> using type = Eigen::Matrix<ScalarType,Rows,1>;
   };
-  template <typename ScalarType> using UnusedVector = Eigen::Matrix<ScalarType,0,1>;
-#define EmptyVector UnusedVector
 
+  /** NullVector<ScalarType>
+   * @brief provides the empty vector (templated by ScalarType)
+   * @concept{vector_concept}
+   */
+  template <typename ScalarType> using NullVector = Eigen::Matrix<ScalarType,0,1>;
+
+  template <typename ScalarType, int Rows> Eigen::Matrix<ScalarType,Rows,1> toEigen(const Eigen::Matrix<ScalarType,Rows,1>& vec) { return vec; }
+
+  /** getRandomVector()
+   * @brief Returns a random vector of the desired type using Eigen::Random()
+   * @concept{vector_concept}
+   */
+  template <template <typename> class VecType>
+  VecType<double> getRandomVector(void) {
+    static_assert(VecType<double>::RowsAtCompileTime>=0,"still need to handle dynamic case");
+    VecType<double> rand(Eigen::Matrix<double,VecType<double>::RowsAtCompileTime,1>::Random());
+    return rand;
+  }
+
+  namespace internal {
+    template<typename VecType, bool Enable = false>
+    struct SizeDispatch {
+      static std::size_t eval(const VecType &vec) { return VecType::RowsAtCompileTime; }
+    };
 
-  // Methods for building complicated vectors out of simpler vectors
-
+    template<typename VecType>
+    struct SizeDispatch<VecType,true > {
+      static std::size_t eval(const VecType &vec) { return vec.size(); }
+    };
+  }
+  /** size()
+   * @brief Evaluate the size of a Vector
+   * @concept{vector_concept}
+   *
+   * @retval RowsAtCompileTime or the result of size() for dynamically sized vectors
+   */
+  template <typename VecType> std::size_t size(const VecType& vec) { return internal::SizeDispatch<VecType,VecType::RowsAtCompileTime==-1>::eval(vec); }
+
+  /** getCoordinateName()
+   * @brief Implements default coordinate names for a generic vector.  Overload this to name your coordinates.
+   * @concept{vector_concept}
+   */
+  template <typename Vector>
+  std::string getCoordinateName(const Vector& vec, unsigned int index) { return "x"+std::to_string(index); }
+
+  /** operator<<()
+   * @brief Implements the ostream operator for your vector types.
+   * @concept{vector_concept}
+   */
+/*
+  template <typename Vector>  // warning: this will match almost anything!
+  std::ostream& operator<<(std::ostream& os, const Vector& vec)
+  {
+    for (int i=0; i<=size(vec); i++)
+      os << getCoordinateName(vec,i) << " = " << vec(i) << std::endl;
+    return os;
+  }
+*/
+  /** CombinedVector<ScalarType,Vector1,Vector2>
+   *
+   * @brief produces a new vector type which is the columnwise composition of vector1 and vector2
+   */
   template <typename ScalarType, template <typename> class Vector1, template <typename> class Vector2>
   class CombinedVector {
   public:
     const static size_t vec1_rows = Vector1<ScalarType>::RowsAtCompileTime,
                         vec2_rows = Vector2<ScalarType>::RowsAtCompileTime;
```

Comment:
Rename to something that makes it clear that this is the number of rows at compile time?